overlay/lib/fetch.nix

```diff
@@ -40,4 +39,34 @@ in
           echo Crate ${name}-${version} not found in ${url}
           exit 1
         '');
+
+  # This implementation of `fetchCrateAlternativeRegistry` assumes that the download URL is updated frequently
+  # on the registry index as a workaround for the lack of authentication for crate downloads. Each time a crate needs
+  # to be downloaded, the whole index needs to be recloned to get the download URL, which is potentially expensive.
```

Comment:
> Each time a crate needs to be downloaded, the whole index needs to be recloned to get the download URL, which is potentially expensive.

Within a single build-run, we can safely assume that the download URL wont change though. Does the implementation take advantage of this?

Reply:
How does it get the URL in the first place?

Reply:
Well, what I meant is that.. when a build starts, it'll clone then index.. and during the build, if it needs to download multiple crates.. one after another.. it can use the same URL for all those downloads. Does that make sense? I'm not sure how it works in this case though.. but I imagine what I just described.

Reply:
And when I said, _build starts_, I meant the build of the project (the root build, not the dependencies).

Reply:
> it'll clone the index.. 

The index has to be each time a crate from that index needs downloading. We can't cache the index because we don't know its SHA beforehand.

Reply:
Let's chat on Slack. I didn't get it properly.